app/controllers/tokens_controller.rb

```diff
@@ -0,0 +1,26 @@
+class TokensController < ApplicationController
+  before_action :require_login
+
+  def new
+    map = current_user.maps.find(params[:map_id])
+    @token = map.tokens.build
+  end
+
+  def create
+    map = current_user.maps.find(params[:map_id])
+    token = map.tokens.new(token_params)
+    if token.save
+      redirect_to map.campaign
+    else
+      redirect_to map.campaign, flash: {
+        error: token.errors.full_messages.to_sentence
+      }
+    end
```

Comment:
Let's add an issue to dismiss the form on success or redraw the form with errors on failure

Reply:
Created https://github.com/jdbann/table/issues/47